substrate/frame/support/src/traits/tokens/fungible/regular.rs

```diff
@@ -187,7 +187,16 @@ pub trait Unbalanced<AccountId>: Inspect<AccountId> {
 			Exact => ensure!(reducible >= amount, TokenError::FundsUnavailable),
 		}
 
-		let new_balance = old_balance.checked_sub(&amount).ok_or(TokenError::FundsUnavailable)?;
+		let new_balance = match old_balance.checked_sub(&amount) {
+			Some(balance) => balance,
+			None => {
+				if matches!(preservation, Preservation::Protect | Preservation::Preserve) {
```

Comment:
Explanation: If the `preservation` is `Protect` or `Preserve` which means the account must be alive after the method call. Hence, if it is failed to `checked_sub` the `old_balance` with the provided `amount` => `BelowMinimum` error thrown